docs/migration-2.0.md

```diff
@@ -0,0 +1,1178 @@
+
+- `PoolCreated` to `ConnectionPoolCreated`
+- `PoolReady` to `ConnectionPoolRead`
+- `PoolCleared` to `ConnectionPoolCleared`
+- `PoolClosedEvent` to `ConnectionPoolClosed`
+- `GetStarted` to `ConnectionCheckOutStarted`
+- `GetFailed` to `ConnectionCheckOutFailed`
+- `GetSucceeded` to `ConnectionCheckedOut`
+- `ConnectionReturned` to `ConnectionCheckedIn`
+
+### CommandFailedEvent
+
+`CommandFailedEvent.Failure` has been converted from a string type to an error type to convey additional error information and to match the type of other `*.Failure` fields, like `ServerHeartbeatFailedEvent.Failure`.
+
+### CommandFinishedEvent
+
+The type for `ServerConnectionID` has been changed to `*int64` to prevent an `int32` overflow and to differentiate between an ID of `0` and no ID.
+
+### CommandStartedEvent
+
+The type for `ServerConnectionID` has been changed to `*int64` to prevent an `int32` overflow and to differentiate between an ID of `0` and no ID.
+
+### ServerHeartbeatFailedEvent
+
+`DurationNanos` has been removed in favor of `Duration`.
+
+### ServerHeartbeatSucceededEvent
+
+`DurationNanos` has been removed in favor of `Duration`.
+
+## Mongo Package
+
+### Client
+
+#### Connect
+
+`Client.Connect` has been removed in favor of `mongo.Connect()`. See section on `NewClient` for more details.
+
+The `context.Context` parameter has been removed from `mongo.Connect()` since [deployment connector](https://github.com/mongodb/mongo-go-driver/blob/a76687682f080c9612295b646a00650d00dd16e1/x/mongo/driver/driver.go#L91) doesn’t accept a context, so the context passed to `mongo.Connect` doesn’t actually do anything.
+
+#### UseSession\[WithOptions\]
+
+This example shows how to construct a session object from a context, rather than using a context to perform session operations.
+
+```go
+// v1
+
+client.UseSession(context.TODO(), func(sctx mongo.SessionContext) error {
+  if err := sctx.StartTransaction(options.Transaction()); err != nil {
+    return err
+  }
+
+  _, err = coll.InsertOne(context.TODO(), bson.D{{"x", 1}})
+
+  return err
+})
+```
+
+```go
+// v2
+
+client.UseSession(context.TODO(), func(ctx context.Context) error {
+  sess := mongo.SessionFromContext(ctx)
+
+  if err := sess.StartTransaction(options.Transaction()); err != nil {
+    return err
+  }
+
+  _, err = coll.InsertOne(context.TODO(), bson.D{{"x", 1}})
+
+  return err
+})
+```
+
+### Collection
+
+#### Clone
+
+This example shows how to migrate usage of the `collection.Clone` method, which no longer returns an error.
+
+```go
+// v1
+
+clonedColl, err := coll.Clone(options.Collection())
+if err != nil {
+  log.Fatalf("failed to clone collection: %v", err)
+}
+```
+
+```go
+// v2
+
+clonedColl := coll.Clone(options.Collection())
+```
+
+#### Distinct
+
+As of v2, the Distinct collection method will return a struct that can be decoded, similar to `Collection.FindOne`. Instead of iterating through an untyped slice, users can decode same-type data using conventional Go syntax.
```

Comment:
```suggestion
The `Distinct()` collection method returns a struct that can be decoded, similar to `Collection.FindOne`. Instead of iterating through an untyped slice, users can decode same-type data using conventional Go syntax.
```